src/readers/morphologyHDF5.cpp

```diff
@@ -182,20 +190,24 @@ bool MorphologyHDF5::_readV2Metadata() {
 
     try {
         HighFive::SilenceHDF5 silence;
-        _file->getGroup(_g_root);
+        _group.getGroup(_g_root);
         _properties._cellLevel._version = MORPHOLOGY_VERSION_H5_2;
         return true;
     } catch (const HighFive::Exception&) {
         return false;
     }
 }
 
-HighFive::DataSet MorphologyHDF5::_getStructureDataSet(size_t nSections) {
+HighFive::DataSet MorphologyHDF5::_getStructureDataSet()
+{
     try {
         HighFive::SilenceHDF5 silence;
-        return _file->getDataSet(_d_structure);
+        return _group.getDataSet(_d_structure);
     } catch (const HighFive::DataSetException&) {
-        return _file->createDataSet<int>(_d_structure, HighFive::DataSpace({nSections, 3}));
+        throw;
+        //FIXME: NO EMPTY DATASET IS CREATED IN CASE THIS ONE IS EMPTY
```

Comment:
Please fix this, still. We may have to have a discussion about how this should be handled.

Reply:
Fixed! Thanks for the help!